### ISSUE TITLE
Dataset unbalance

### ISSUE THREAD

**Issue by shpotes** [shpotes/tensorflowers#16]
Nuestro dataset de entrenamiento está bastante desbalanceado y esto esta haciendo que las predicciones colapsen a un conjunto pequeño de etiquetas.
Se me ocurren diferentes formas de resolverlo:
- [x] #18
- [x] #19
- [x] #20
- [x] Podemos entrenar el modelo en un procedimiento de dos faces: 
    1. Dataset con batches balanceados, para esto se debe crear una `collate_fn` que samplee uniformemente de cada uno de las clases (si la clase se queda sin nuevos datos simplemente se repideten los ya usados), este paso sería equivalente a entrenar usando oversampling. 
    2.  Fine tunear el modelo con el dataset original un par de epochs para que este se acostumbre a la distribucion dispar de los labels

**Comment by shpotes** [comment]
@Scarta19 y @aospinap1 podría jugar al rededor del oversampling, para esto deberían encontrar un conjunto de augmentation interesantes y tratar de aplicarlas, en [kornia](https://kornia.readthedocs.io/en/latest/augmentation.html) hay un par de augmentations interesantes.

@MaximumPatronus podría tratar de balancear la funcion de coste, usando el [API de scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.utils.class_weight.compute_class_weight.html) (CrossEntropyLoss recive weight como parametro).

@shpotes podría tratar de ajustar la implementacion de `timm` de ASL a nuestro codebase.
